projects/media-viewer/src/lib/viewers/image-viewer/image-viewer.component.ts

```diff
@@ -115,10 +122,69 @@ export class ImageViewerComponent implements OnChanges {
   }
 
   public updateAnnotation(updatedAnnotation: Annotation) {
-    const annotations = this.annotationSet.annotations
-      .filter(annotation => annotation.id !== updatedAnnotation.id);
+    const annotations = this.annotationSet.annotations.filter(annotation => annotation.id !== updatedAnnotation.id);
+
     annotations.push(updatedAnnotation);
     this.annotationSet.annotations = annotations;
     this.api.postAnnotationSet(this.annotationSet);
   }
+
+  public onMouseDown(event: MouseEvent) {
```

Comment:
should this happen here or in the rectangle component?

Reply:
this will all go into the annotation set component.... as soon as we get one